### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by StupidCodeThief**

### CODE REVIEW

File: api/validation/auth.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  if (emailValidate.error) {
+    errors.email = emailValidate.error.details[0].message.replace(
+      `"value"`,
+      "Email"
+    );
+  }
+
+  if (passwordValidate.error) {
+    if (passwordValidate.error.details[0].context.regex) {
+      errors.passwor = "Password must include at least one didgit";
```
Comment (StupidCodeThief) [inline api/validation/auth.js]: Already change it

File: api/services/errorHandlers/tokenNotValidError.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const ApplicationError = require("./applicationError");
+
+class TokenNotValid extends ApplicationError {
+  constructor() {
+    super("Token not valid", 404);
+  }
+}
```
Comment (slavasgod) [inline api/services/errorHandlers/tokenNotValidError.js]: It isn't a 404 error

File: api/services/errorHandlers/unauthorizedError.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const ApplicationError = require("./applicationError");
+
+class UnauthorizedError extends ApplicationError {
+  constructor() {
+    super("Unauthorized", 404);
+  }
+}
```
Comment (slavasgod) [inline api/services/errorHandlers/unauthorizedError.js]: It isn't a 404 error

File: api/controllers/auth.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      return res.status(400).json(errors);
+    }
+
+    const response = await authService.getUser(req.user);
+    res.status(201).send(response);
+  } catch (error) {
+    console.error(error.message);
+    res.status(500).send("Server error");
+  }
+};
```
Comment (slavasgod) [inline api/controllers/auth.js]: Why you validate token with joi? You already has jwt guard middleware, that verifies you token

File: api/services/auth.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+const getUser = async ({ id, email }) => {
+  try {
+    const user = await User.findOne({ where: { email } });
+
+    if (!user) {
+      return new errors.NotFoundError("User not found");
+    }
+
+    user.password = undefined;
+    return user;
```
Comment (slavasgod) [inline api/services/auth.js]: Use sql queries to select what you want to get from database

File: api/services/errorHandlers/userAlreadyExistsError.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const ApplicationError = require("./applicationError");
+
+class UserAlreadyExists extends ApplicationError {
+  constructor(message) {
+    super(message || 'User already exists.', 404);
+  }
+}
```
Comment (slavasgod) [inline api/services/errorHandlers/userAlreadyExistsError.js]: It isn't a 404 error

**Review by slavasgod — CHANGES_REQUESTED** [review]

File: api/services/auth.js (hunk 5 above)
Comment (StupidCodeThief) [reply api/services/auth.js]: done

File: api/services/errorHandlers/tokenNotValidError.js (hunk 2 above)
Comment (StupidCodeThief) [reply api/services/errorHandlers/tokenNotValidError.js]: done, change to 403

File: api/services/errorHandlers/unauthorizedError.js (hunk 3 above)
Comment (StupidCodeThief) [reply api/services/errorHandlers/unauthorizedError.js]: done, change to 401

File: api/controllers/auth.js (hunk 4 above)
Comment (StupidCodeThief) [reply api/controllers/auth.js]: done, remove token validation at controller

File: api/services/errorHandlers/userAlreadyExistsError.js (hunk 6 above)
Comment (StupidCodeThief) [reply api/services/errorHandlers/userAlreadyExistsError.js]: done, change to 403

**Review by slavasgod — APPROVED** [review]

**Pull request merged**